docs/02-skog-context.md

```diff
@@ -0,0 +1,111 @@
+# Async Context in Skog
+
+In this page you will learn more on how Skog uses Async Context and how the functions `runWithSkogContext` and `getFields` can help you to make integrations using its power.
+
+- Both functions handle **Node.js Async Context**. If you don't know what it is, you can read about it [here](./01-nodejs-context.md).
+- This page is not the API reference of the functions but an explanation of them. [ todo: create such page and add a link to it here ]
+
+---
+
+- The function **runWithContext** receives two arguments: `fields` (an object) and `callback` (a zero-argument function).
+
+  It will create a new context with `fields`. The context will be alive during the execution of the `callback`.
```

Comment:
Replaced `callback` with `fn`. However I still use the word "callback" as term (not as variable name) to refer to it in some text below.

I have also added an example explaining the benefit of returning the same type as `fn`

---

I mean this:

```
// the "T" returned  by `fn` = "T" returned by `runWithSkogContext`
runWithSkogContext<T>(fields: object, fn: () => T): T;
```

(sometimes is easier to explain with types rather than with text :D)

Reply:
@exacs I still think you should drop callback entirely. Since callback has a lot of meaning, especially in Node.js, it could cause confusion. I think it is okay to just call it "the function" or "the function to be executed". See how thet talk about function params when explaining higher order functions here https://medium.com/javascript-scene/higher-order-functions-composing-software-5365cf2cbe99

Reply:
I've updated both documents to avoid the word "callback". I've looked at several sources and while "callback" just means "a function you pass as a parameter so something else calls it", it is true that in the JavaScript/Node.js it has more connotations.

Both `.md` files are modified:

- https://github.com/KTH/skog/pull/16/commits/08223ac08684d51be66fc36879ecef10d2f2246b
- https://github.com/KTH/skog/pull/16/commits/458bf1326d39f1143f2427b87996dfb79d36455b